dependent_type_theory.md

```diff
@@ -803,8 +824,10 @@ commands such as ``set_option`` and ``open``.
 In many respects, however, a ``namespace ... end`` block behaves the
 same as a ``section ... end`` block. In particular, if you use the
 ``variable`` command within a namespace, its scope is limited to the
-namespace. Similarly, if you use an ``open`` command within a
-namespace, its effects disappear when the namespace is closed.
+namespace (variables are also private inside a namespace). Similarly,
```

Comment:
Comment above about "private" is also applicable here.

Reply:
Hmm, what I'm trying to convey here is that even if I do "open namespace" I cannot reference any variables defined in that namespace, it is as if variables are automatically "private", no?

Reply:
I see, so this doesn't work which is very interesting/odd.  So variable scope is limited to the one namespace block that they are defined in.  Is that a more accurate way of stating it?

```lean
namespace foo
variable (alpha : Type)
end foo

namespace foo
def doTwice (h : alpha → alpha) (x : alpha) : alpha :=
  h (h x)
end foo
```

so I added more verbage to attempt to make this more clear...

Reply:
@lovettchris The `variable` command is essentially `local` in scope and thus goes out of existence once the scope closes (just like a `local` `syntax` or `macro`). In contrast, if they continued exist whenever the namespace was opened (either via `open Foo` or `namespace Foo`), that would be more akin to a `scoped` `syntax` or `macro`.

Reply:
Ok, so is this better?

In particular, if you use the
``variable`` command within a namespace, its scope is limited to the
one namespace block they are defined in, which means they are not
accessible in another namespace block of the same name and they
are not accessible if you `open` the namespace later.  They behave
like a local variable to the specific namespace block.

Reply:
@lovettchris looks fine to me! (other than the typo note of mixing singular and plural -- 'its scope' vs 'they are defined')

Reply:
good catch, fixed.